recipes-soletta/soletta/soletta_git.bb

```diff
@@ -79,6 +80,54 @@ do_configure_prepend() {
 }
 
 do_compile() {
+   # changing the home directory to the working directory, the .npmrc will be created in this directory
+   export HOME=${WORKDIR}
+
+   # does not build dev packages
+   npm config set dev false
+
+   # access npm registry using http
+   npm set strict-ssl false
+   npm config set registry http://registry.npmjs.org/
+
+   # configure http proxy if neccessary
+   if [ -n "${http_proxy}" ]; then
+       npm config set proxy ${http_proxy}
+       NODE_GYP_PROXY="--proxy=${http_proxy}"
+   fi
+   if [ -n "${HTTP_PROXY}" ]; then
+       npm config set proxy ${HTTP_PROXY}
+       NODE_GYP_PROXY="--proxy=${HTTP_PROXY}"
+   fi
+
+   # configure cache to be in working directory
+   npm set cache ${WORKDIR}/npm_cache
+
+   # clear local cache prior to each compile
+   npm cache clear
+
+   case ${TARGET_ARCH} in
+       i?86) targetArch="ia32"
+           echo "targetArch = 32"
+           ;;
+       x86_64) targetArch="x64"
+           echo "targetArch = 64"
+           ;;
+       arm) targetArch="arm"
+           ;;
+       mips) targetArch="mips"
+           ;;
+       sparc) targetArch="sparc"
+           ;;
+       *) echo "unknown architecture"
+          exit 1
+           ;;
+   esac
```

Comment:
We pass targetArch value to node-gyp as --arch option, but echo is just for debugging the shell task. It echo the text to stdout in verbose mode and also stores in the log file.